lib/plugins/aws/provider/awsProvider.js

```diff
@@ -89,19 +89,31 @@ const impl = {
         });
       };
 
-      const profileCredentials = new AWS.SharedIniFileCredentials(params);
-      if (
-        !(
-          profileCredentials.accessKeyId ||
-          profileCredentials.sessionToken ||
-          profileCredentials.roleArn
-        )
-      ) {
-        throw new Error(`Profile ${profile} does not exist`);
-      }
-
-      impl.addCredentials(results, profileCredentials);
+      const profileCredentials = [
+        BbPromise.resolve(new AWS.SharedIniFileCredentials(params)),
+        new BbPromise(resolve => {
+          let credential;
+          params.callback = err => {
+            return err
+              ? resolve({}) // should not reject, since process creds are optional
+              : resolve(credential);
+          };
+          credential = new AWS.ProcessCredentials(params);
```

Comment:
It'll be nice to fallback to this check only when `new AWS.SharedIniFileCredentials(params)` doesn't return credentials.